psydac/linalg/basic.py

```diff
@@ -900,10 +928,39 @@ def dtype(self):
     @property
     def linop(self):
         return self._A
+    
+    @linop.setter
+    def linop(self, a):
+        assert isinstance(a, LinearOperator)
+        assert a.domain is self.domain
+        assert a.codomain is self.codomain
+        self._A = a
 
     @property
     def options(self):
         return self._options
+    
+    @property
+    @abstractmethod
+    def solver(self):
+        "String that identifies the solver."
+        pass
```

Comment:
We don't need this really, it is just for convenience. In case you have to check for the type of the solver, now you can just call `A_inv.solver in ('cg', 'pcg')` for example. With just the class names this would turn into sg like `A_inv.__name__ in ('ConjugateGradient', 'PConjugateGradient')`. But we can remove, as you prefer.

Reply:
Why would one check the solver that they just created? For unit testing or debugging I would prefer using the classes, albeit more verbose, with
```python
if isinstance(A_inv, (ConjugateGradient, PConjugateGradient)):
    ...
```

Reply:
So yes, I would remove the `solver` property for now.